packages/fx-core/src/common/tools.ts

```diff
@@ -948,3 +950,39 @@ export async function getSPFxToken(
   }
   return spoToken;
 }
+
+// Get some fields from project settings and will be used in telemetry
+export function getProjectSettingsForCommonTelemetry(rootPath: string | undefined): any {
+  if (!rootPath) {
+    return undefined;
+  }
+
+  try {
+    const projectSettingsPath = path.join(
+      rootPath,
+      `.${ConfigFolderName}`,
+      InputConfigsFolderName,
+      ProjectSettingsFileName
+    );
+
+    if (!projectSettingsPath || !fs.pathExistsSync(projectSettingsPath)) {
+      return undefined;
+    }
+
+    const projectSettings = fs.readJsonSync(projectSettingsPath);
+    return {
+      projectId: projectSettings.projectId ?? undefined,
+      version: projectSettings.version ?? undefined,
+      isFromSample: projectSettings.isFromSample ?? undefined,
+      solutionSettings: projectSettings.solutionSettings
+        ? JSON.stringify(projectSettings.solutionSettings)
+        : undefined,
+      programmingLanguage: projectSettings.programmingLanguage ?? undefined,
+      pluginSettings: projectSettings.pluginSettings
+        ? JSON.stringify(projectSettings.pluginSettings)
```

Comment:
pluginSettings may contains csproj file path. Will that cause the whole pluginSettings be suppressed?

Reply:
Should we whitelist fields for `pluginSettings`? 

If so, `host-type` and `capabilities` in bot plugin should be whitelisted.

```
    "pluginSettings": {
        "fx-resource-bot": {
            "host-type": "app-service",
            "capabilities": [
                "notification"
            ]
        }
    },
```